### DISCUSSION TITLE
Can You Set Your Connected Wallet as Default Provider?

### DISCUSSION THREAD

**Discussion by johnpaulcas** [wevm/wagmi#2217] [Q&A]
Is it currently possible to set a connected wallet as the default provider on wagmi? For instance, if I connect my Metamask wallet, can it automatically become my default provider instead of the Alchemy API ID that I provided?

**Comment by jxom** [comment]
It is possible, but it is very highly not recommended. Some popular browser extension wallets & WalletConnect v2 wallet configurations do not support all public RPC methods that are required for the functionality of wagmi such as `eth_call`, etc.
